### ISSUE TITLE
Negative mass on parts.

### ISSUE THREAD

**Issue by Lisias** [net-lisias-ksp/TweakScale#11]
Some parts are getting negative mass when scaled down. Investigate.

See : 
https://forum.kerbalspaceprogram.com/index.php?/topic/180017-engine-has-negative-mass/&do=findComment&comment=3492469

Note: this was happening on 1.3.1 , so, no novelty. KSP versions are not the problem.

**Comment by Lisias** [comment]
Possible relation with #9

**Comment by Lisias** [comment]
Occurence 1: KSP 1.3.1 and:

000_AT_Utils - 1.5
USI Tools - 0.10.2
ToolbarControl - 0.1.4.7
Ablative-Airbrake - 0.3
AlmostFreeLaunchClamps - 0.2.2
B9 Animation Modules - 1.1
B9 Part Switch - 2.1.1﻿
B9 Aerospace HX Parts - 6.4
﻿ CameraTools - 1.9
CCTV - 1.1.2
Chatterer Extended - 0.6.2
Chatterer - 0.9.93.1804
CollisionFX - 4.0
Community Category Kit - 2.0.2
Community Resource Pack - 0.8.1
CommunityTechTree - 3.2.1
CryoEngines - 0.5.11
CryoTanks - 0.4.9
Deadly Reentry - 7.6.2
DynamicBatteryStorage - 1.2
FASA Launch Clamps - 1.1.4
Ferram Aerospace Research - 0.15.9.1
Firespitter - 7.6
FMRS (Flight Manager For Reusable Stages) - 1.2.6
GroundConstruction - 1.2.1
HullcamVDSContinued - 0.1.8
JSIPartUtilities - 0.4.1
RasterPropMonitor - 0.29.3
Kerbal Engineer Redux - 1.1.3
Kerbal Joint Reinforcement - 3.3.3
KerbalLaunchFailure - 0.4.8
HyperEdit - 1.5.6
<b><color=#CA7B3C>Kopernicus</color></b> - 1.3.1.7
KSP-AVC Plugin - 1.1.6.2
KWRocketryRedux - 3.1.4
MagiCore - 1.3.1
Infernal Robots - 2.0.14
ModularFlightIntegrator - 1.2.4
Monthly Budgets - 4.1.0.1
NASA_CountDown - 1.3.6.2
NearFutureConstruction - 0.8.4
NearFutureLaunchVehicles - 1.1.4
NearFutureProps - 0.3
NEBULA_DecalsContinued - 0.1.2
Procedural Parts - 1.2.15
RealChute - 1.4.5
RealFuels - 12.6
RealFuels Stockalike Configs - 3.2.3
RealSolarSystem - 13.1
Recovery Controller - 0.0.2
US and Soviet Solar Panels Pack - 1.5
RSSDateTime - 1.2
SCANsat - 1.1.8.4
SolverEngines - 3.5
StationPartsExpansionRedux - 1.0.2
TAC Life Support - 0.13.6
TweakScale - 2.3.7
USI Core - 0.5.1
Konstruction - 0.3.1
MKS - 0.53.2
VenStockRevamp - 1.9.5

**Comment by Lisias** [comment]
A **possible** explanation for this behaviour. Untested. Pure guess. But it worths some tests on the next time window.

https://forum.kerbalspaceprogram.com/index.php?/topic/140541-151-b9partswitch-v251-december-1/&page=16&tab=comments#comment-3499052

**Comment by Lisias** [comment]
Another issue. B9 Part Switch appears to be related only due popularity. (it yet another issue)

https://forum.kerbalspaceprogram.com/index.php?/topic/179030-141151-tweakscale-under-new-management-2018-1127/&do=findComment&comment=3501824

https://forum.kerbalspaceprogram.com/index.php?/topic/179030-141151-tweakscale-under-new-management-2018-1127/&do=findComment&comment=3502116

**Comment by Lisias** [comment]
Issue #13 should handle at least some of the problematic parts described here.

TODO: check them to see if every one of them are ModulePartVariants .

**Comment by Lisias** [comment]
Need help to confirm that his issue is directly related to #13 or if this is a new problem.

**Comment by Lisias** [comment]
Nope, I just confirmed. This issue is, in fact, directly related to #13 and #9 .

Closing this as DUP.

**Comment by Lisias** [comment]
Just to document that this is, indeed, happening on the field.

https://github.com/pellinor0/TweakScale/issues/83

**Comment by snakeru** [comment]
Hi @Lisias. So apparently, this is a known issue :)
No complaints, I'm just happy there is more info available to troubleshoot it.

You asked about the KSP.log, here it is. Thanks!
[KSP.log](https://github.com/net-lisias-ksp/TweakScale/files/2809543/KSP.log)

**Comment by Lisias** [comment]
Yeah, this is happening again. This time, on the KSP installment from the fellow Kerbonaut [Einstein_Cross_X1](https://forum.kerbalspaceprogram.com/index.php?/profile/175674-einstein_cross_x1/).

Evidences (KSP.log, ModuleManager cache, Add'Ons list):
* [Tweakscale Error.zip](https://github.com/net-lisias-ksp/TweakScale/files/3105234/Tweakscale.Error.zip)

**Comment by spacegeek37** [comment]
I'm not sure if this helps, but I've encountered the same error, both months ago and now. I've narrowed down the problem to these mods:

Real Fuels (stockalike config)
Real Fuels
Solver Engines
Tweakscale
Module Manager
Community Resource Pack

All are the most recent versions available on CKAN running on KSP 1.7.2.

Also worth noting is that this only happens when I scale down (not up) parts with Real Fuels set up, so KSPIE engines and non-engine parts work fine. Here is the log:

[ModuleManager.zip](https://github.com/net-lisias-ksp/TweakScale/files/3332445/ModuleManager.zip)

I tried using the Real Fuels configs to change the mass of the RK-7 back to its stock value on a different save, but it simply made the negative mass larger.

**Comment by Lisias** [comment]
Recently, KSP started to clamp the Mass when third-parties tries to set it.

However, since TweakScale still aims to support older KSP versions, this still needs to be tackled down somehow.

Unfortunately, this will not happen before TweakScale 2.5, and perhaps it will ended up being postponed to TweakScale3.

**Comment by Lisias** [comment]
Moving this one to the 2.5.0.x series, following the [ROAD MAP](https://github.com/net-lisias-ksp/TweakScale/blob/master/ROAD_MAP.md).
